reference-implementation/lib/transform-stream.js

```diff
@@ -16,6 +17,17 @@ function TransformStreamCloseReadable(transformStream) {
     throw new TypeError('Readable side is already closed');
   }
 
+  if (transformStream._writableDone === true) {
+    throw new TypeError('TransformStream is already closing');
+  }
```

Comment:
Did you try to limit the way to close the readable side to only by resolving flush() once flush() is invoked?  What's the motivation for that?